cid/crud.py

```diff
@@ -106,23 +106,39 @@ def latest_azure_image(db: Session, arch: Optional[str]) -> dict[str, Any]:
     return latest_images_dict
 
 
-def latest_google_image(db: Session) -> dict:
+def latest_google_image(db: Session, arch: Optional[str]) -> dict:
     """Get the latest RHEL image on Google Cloud."""
-    latest_image = (
-        db.query(GoogleImage)
-        .order_by(desc(GoogleImage.version), desc(GoogleImage.creationTimestamp))
-        .first()
+    archs = (
+        [arch]
+        if arch is not None
+        else [arch[0] for arch in db.query(GoogleImage.arch).distinct().all()]
     )
+    print(archs)
+    latest_images_dict = {}
+    for arch in archs:
+        print(arch)
+        latest_image = (
+            db.query(GoogleImage)
+            .filter(GoogleImage.arch.ilike(arch))
```

Comment:
This is interesting.
sqalchemy queries seem to be case sensitive by default  (which is expected).
running a query like this will work: 
"localhost:8000/google/latest?arch=ARM64"

but if I run this:
"localhost:8000/google/latest?arch=X86_64"
it will automatically be tranformed to "...x86_64"  which will crash.
Used ilike comparison to mitigate.